src/core/ext/client_channel/parse_address.c

```diff
@@ -49,12 +49,11 @@
 
 int parse_unix(grpc_uri *uri, grpc_resolved_address *resolved_addr) {
   struct sockaddr_un *un = (struct sockaddr_un *)resolved_addr->addr;
-
+  const size_t maxlen = sizeof(un->sun_path);
+  if (strnlen(uri->path, maxlen) == maxlen) return 0;
   un->sun_family = AF_UNIX;
-  strncpy(un->sun_path, uri->path, sizeof(un->sun_path) - 1 /* null term'd */);
-  un->sun_path[sizeof(un->sun_path) - 1] = '\0';
-  resolved_addr->len = strlen(un->sun_path) + sizeof(un->sun_family) + 1;
-
+  strcpy(un->sun_path, uri->path);
+  resolved_addr->len = strlen(un->sun_path) + maxlen + 1;
```

Comment:
Looks like you put `maxlen` here instead of `sizeof(un->sun_family)`, where you probably actually meant to `replace sizeof(un->sun_path)`.

Shouldn't this actually just be set to `sizeof(struct sockaddr_un)`?